### PR TITLE
Upgrade Rails and Mongoid

### PR DESCRIPTION
**Pull request by benlovell**
Trello: https://trello.com/c/xRT0XUs5

f38601d will be squashed after review.

Has the requisite PRs in puppet to configure both `SECRET_KEY_BASE` and `MONGODB_URI`.

### CODE REVIEW

**Comment by jamiecobbett** [comment]
I know you don't have access to it, but we'll need to do something about [the mongoid.yml that overwrites this on deploy](https://github.gds/gds/alphagov-deployment/blob/master/govuk_need_api/to_upload/mongoid.yml)  as that uses the old format.

The simplest way of doing this is to copy [asset-manager](https://github.com/alphagov/asset-manager/blob/master/config/mongoid.yml), first making the corresponding changes in [puppet](https://github.gds/gds/puppet/pull/3596). Once everything is deployed, the mongoid.yml in alphagov-deployment can be removed.

File: config/environments/test.rb [hunk 1]
```diff
@@ -8,28 +8,44 @@
...
   # Log error messages when you accidentally call methods on nil
   config.whiny_nils = true
 
   # Show full error reports and disable caching
+  # Show full error reports and disable caching.
   config.consider_all_requests_local       = true
   config.action_controller.perform_caching = false
 
   # Raise exceptions instead of rendering exception templates
+  # Raise exceptions instead of rendering exception templates.
```
Comment (jamiecobbett) [inline config/environments/test.rb]: Duplicate comment

File: config/environments/test.rb [hunk 2]
```diff
@@ -8,28 +8,44 @@
...
+
+  # Configure static file server for tests with Cache-Control for performance.
+  config.serve_static_files   = true
+  config.static_cache_control = 'public, max-age=3600'
 
   # Log error messages when you accidentally call methods on nil
   config.whiny_nils = true
 
   # Show full error reports and disable caching
+  # Show full error reports and disable caching.
```
Comment (jamiecobbett) [inline config/environments/test.rb]: Duplicate comment

File: config/secrets.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+development:
+  secret_key_base: '274b52c8af8dbcf6dca835e45b607244d76fd28501e873d477daaf1cbc450645434d1cf4b7935df3dcd859ae46a4dadcd26bd22eb506eed28fffefb688651989'
+
+test:
+  secret_key_base: '274b52c8af8dbcf6dca835e45b607244d76fd28501e873d477daaf1cbc450645434d1cf4b7935df3dcd859ae46a4dadcd26bd22eb506eed28fffefb688651989'
+
+production:
+  secret_key_base: <%= ENV['SECRET_KEY_BASE'] %>
```
Comment (jamiecobbett) [inline config/secrets.yml]: This will need to be populated from puppet and deployment, like we want to do the mongoid URL.

Comment (benlovell) [inline config/secrets.yml]: I have a PR outstanding as per the PR comment.

**Comment by benlovell** [comment]
@jamiecobbett the comments have been addressed and rebased in. I've also got the two PRs pending for puppet that @boffbowsh is dealing with for production hieradata and removal of the existing `mongoid.yml` that would overwrite this project's.

**Comment by jamiecobbett** [comment]
I've opened an [alphagov-deployment PR to remove mongoid.yml](https://github.gds/gds/alphagov-deployment/pull/1119) when this PR is ready to merge.

**Pull request merged**